create-release/index.js

```diff
@@ -37,15 +37,20 @@ async function main() {
         const env = getEnvironment(tagSplitted[2]);
         previousTag = getPreviousTagWithEnv(allTags.data, env);
     }
-    
-    console.log('Previous tag', previousTag.name)
-    console.log('tag',github.context.payload.ref)
-    const releaseNoteGenerated = await octokit.rest.repos.generateReleaseNotes({
+
+    let dataReleaseNoteGenerated = {
         owner,
         repo,
         tag_name: github.context.payload.ref,
-        previous_tag_name: previousTag.name
-    })
+    };
+
+    if (previousTag) {
+        dataReleaseNoteGenerated.previous_tag_name = previousTag?.name;
+        console.log('Previous tag', previousTag?.name)
```

Comment:
No need for optional chaining here, you already check the variable in the if condition.
```suggestion
        dataReleaseNoteGenerated.previous_tag_name = previousTag.name;
        console.log('Previous tag', previousTag.name)
```